akka-docs/src/main/paradox/scala/stream/stream-io.md

```diff
@@ -1,29 +1,37 @@
 # Working with streaming IO
 
 Akka Streams provides a way of handling File IO and TCP connections with Streams.
-While the general approach is very similar to the @ref:[Actor based TCP handling](../io-tcp.md) using Akka IO,
+While the general approach is very similar to the @ref[Actor based TCP handling](../io-tcp.md) using Akka IO,
 by using Akka Streams you are freed of having to manually react to back-pressure signals,
 as the library does it transparently for you.
 
 ## Streaming TCP
 
 ### Accepting connections: Echo Server
 
-In order to implement a simple EchoServer we `bind` to a given address, which returns a `Source[IncomingConnection, Future[ServerBinding]]`,
+In order to implement a simple EchoServer we `bind` to a given address, which returns a @scala[`Source[IncomingConnection, Future[ServerBinding]]`]@java[`Source<IncomingConnection, CompletionStage<ServerBinding>>`],
 which will emit an `IncomingConnection` element for each new connection that the Server should handle:
 
-@@snip [StreamTcpDocSpec.scala]($code$/scala/docs/stream/io/StreamTcpDocSpec.scala) { #echo-server-simple-bind }
+Scala
+:   @@snip [StreamTcpDocSpec.scala]($code$/scala/docs/stream/io/StreamTcpDocSpec.scala) { #echo-server-simple-bind }
+
+Java
+:   @@snip [StreamTcpDocTest.java]($code$/java/jdocs/stream/io/StreamTcpDocTest.java) { #echo-server-simple-bind }
 
 ![tcp-stream-bind.png](../../images/tcp-stream-bind.png)
 
 Next, we simply handle *each* incoming connection using a `Flow` which will be used as the processing stage
 to handle and emit `ByteString` s from and to the TCP Socket. Since one `ByteString` does not have to necessarily
-correspond to exactly one line of text (the client might be sending the line in chunks) we use the `Framing.delimiter`
-helper Flow to chunk the inputs up into actual lines of text. The last boolean
+correspond to exactly one line of text (the client might be sending the line in chunks) we use the @scala[`Framing.delimiter`]@java[`delimiter`]
+helper Flow @scala[]@java[from `akka.stream.javadsl.Framing`] to chunk the inputs up into actual lines of text. The last boolean
```

Comment:
Perhaps here "@scala[`Framing.delimiter helper Flow`]@java[`delimiter` helper Flow from `akka.stream.javadsl.Framing`]" might be clearer